hw/drivers/sensors/lps33hw/src/lps33hw.c

```diff
@@ -1025,59 +1080,124 @@ static int
 lps33hw_sensor_read(struct sensor *sensor, sensor_type_t type,
         sensor_data_func_t data_func, void *data_arg, uint32_t timeout)
 {
-    (void)timeout;
-    int rc = SYS_EINVAL;
-    struct sensor_itf *itf;
-
-    itf = SENSOR_GET_ITF(sensor);
-    if (type & SENSOR_TYPE_PRESSURE) {
-        struct lps33hw *lps33hw;
-
-        lps33hw = (struct lps33hw *)SENSOR_GET_DEVICE(sensor);
-        if (lps33hw->cfg.int_cfg.data_rdy) {
-            /* Stream read */
-            lps33hw->pdd.user_handler = data_func;
-            lps33hw->pdd.user_arg = data_arg;
-
-            rc = lps33hw_enable_interrupt(sensor,
-                lps33hw_read_interrupt_handler, sensor);
-            if (rc) {
-                return rc;
-            }
-        } else {
-            /* Read once */
-            struct sensor_press_data spd;
-
-            rc = lps33hw_get_pressure(itf, &spd.spd_press);
-            if (rc) {
-                return rc;
-            }
-
-            spd.spd_press_is_valid = 1;
-
-            rc = data_func(sensor, data_arg, &spd, SENSOR_TYPE_PRESSURE);
-        }
+  int rc = SYS_EINVAL;
+#if MYNEWT_VAL(LPS33HW_ONE_SHOT_MODE)
+  struct sensor_itf *itf;
+  itf = SENSOR_GET_ITF(sensor);
+  uint8_t rate;
+  rc = lps33hw_get_value(itf, LPS33HW_CTRL_REG1_ODR, &rate);
+  if (rc) {
+      return rc;
+  }
+  struct lps33hw *lps33hw;
+  lps33hw = (struct lps33hw *)SENSOR_GET_DEVICE(sensor);
+  if(rate == LPS33HW_75HZ){
+      if (type & SENSOR_TYPE_PRESSURE) {
+	  if (lps33hw->cfg.int_cfg.data_rdy) {
+	      /* Stream read */
+	      lps33hw->pdd.user_handler = data_func;
+	      lps33hw->pdd.user_arg = data_arg;
+	      rc = lps33hw_enable_interrupt(sensor,
+	            lps33hw_read_interrupt_handler, sensor);
+	      if (rc) {
+		  return rc;
+	      }
+	  }
+	  else {
+	      /* Read once */
+	      struct sensor_press_data spd;
+	      rc = lps33hw_get_pressure(itf, &spd.spd_press);
+	      if (rc) {
+		  return rc;
+	      }
+	      spd.spd_press_is_valid = 1;
+	      rc = data_func(sensor, data_arg, &spd, SENSOR_TYPE_PRESSURE);
+	  }
+	  if (type & SENSOR_TYPE_TEMPERATURE) {
+	      struct sensor_temp_data std;
+	      rc = lps33hw_get_temperature(itf, &std.std_temp);
+	      if (rc) {
+		  return rc;
+	      }
+	      std.std_temp_is_valid = 1;
+	      rc = data_func(sensor, data_arg, &std,
+			     SENSOR_TYPE_TEMPERATURE);
+	  }
+	  if (!(type & SENSOR_TYPE_TEMPERATURE) && !(type & SENSOR_TYPE_PRESSURE)) {
+	      return SYS_EINVAL;
+	  }
+      }
```

Comment:
this closing } was in different place in before, now it's not possible to read temperature without pressure